src/views/transactions/components/account-selector/account-selector.view.tsx

```diff
@@ -0,0 +1,203 @@
+import React from "react";
+import { TxType } from "@hermeznetwork/hermezjs/src/enums";
+
+import AccountList from "src/views/shared/account-list/account-list.view";
+import Container from "src/views/shared/container/container.view";
+import InfiniteScroll from "src/views/shared/infinite-scroll/infinite-scroll.view";
+import Spinner from "src/views/shared/spinner/spinner.view";
+import useAccountSelectorStyles from "src/views/transactions/components/account-selector/account-selector.styles";
+import { AsyncTask } from "src/utils/types";
+import { Pagination } from "src/utils/api";
+// domain
+import {
+  HermezAccount,
+  EthereumAccount,
+  FiatExchangeRates,
+  PoolTransaction,
+  PendingDeposit,
+  isHermezAccount,
+  isEthereumAccount,
+} from "src/domain/hermez";
+
+export interface AccountsWithPagination {
+  accounts: HermezAccount[];
+  pagination: Pagination;
+}
+
+type Transaction =
```

Comment:
Instead of having a prop `transaction` with this type I would suggest to type the `AccountSelectorProps` directly using the union as it feels more natural to have more props instead of having one prop `transaction` with all the props and the handlers.

As we have now:

```ts
              <AccountSelector
                transaction={{
                  type: TxType.Deposit,
                  accountsTask: ethereumAccountsTask,
                  pendingDeposits: accountPendingDeposits,
                  onLoadAccounts: onLoadEthereumAccounts,
                  onAccountClick: onGoToBuildTransactionStep,
                }}
                preferredCurrency={preferredCurrency}
                fiatExchangeRates={
                  fiatExchangeRatesTask.status === "successful" ||
                  fiatExchangeRatesTask.status === "reloading"
                    ? fiatExchangeRatesTask.data
                    : {}
                }
              />
```

As proposed:

```ts
              <AccountSelector
                txType={TxType.Deposit}
                accountsTask={ethereumAccountsTask}
                pendingDeposits={accountPendingDeposits}
                preferredCurrency={preferredCurrency}
                fiatExchangeRates={
                  fiatExchangeRatesTask.status === "successful" ||
                  fiatExchangeRatesTask.status === "reloading"
                    ? fiatExchangeRatesTask.data
                    : {}
                }
                onLoadAccounts={onLoadEthereumAccounts}
                onAccountClick={onGoToBuildTransactionStep}
              />
```

As you can see types fit better with React.

Reply:
Done. Very good point. Thanks.

I will apply this change also to `transaction-overview.view.tsx`.

Reply:
Done also for `TransactionOverviewProps`. Much better!

Reply:
Yeah! Nice :)